Publications and news

kidSoba opened the pull request:
This PR adds:

- several refactors
- publications page to homepage
- temporary news section (to be updated with a better solution later; currently hard coded)

kidSoba committed:
build: add various icons and images

kidSoba committed:
feat: add homepage and several other pages + misc

adds homepage and sets it as page to be directed from "home"; add overview pages, as well as links to publications (still old open-ease); also adds nav-bar highlighting

kidSoba committed:
refactor: refactor image name in QA.html

kidSoba committed:
feat: add link to banner

kidSoba committed:
fix: adjust layout-paddings

kidSoba committed:
build: remove deprecated files for overview-functionality

kidSoba committed:
build: add new dependencies

kidSoba committed:
refactor: move password_validator() to utility-module

kidSoba committed:
refactor: set neem default-image during initilization

kidSoba committed:
feat: add 'last_updated'-attribute to neems

kidSoba committed:
feat: add "is_featured"-attribute to neems

this attribute is used to determine if the neem will featured at the top of the homepage

kidSoba committed:
feat: add scheduler task to load markdown-files for visible neems

kidSoba committed:
refactor: load homepage neem-card information from neemhub

fetch the featured and most recent neems from the neemhub for the homepage cards

kidSoba committed:
feat: add overview-pages created from repo-markdowns

kidSoba committed:
feat: refactor code & add feature to download and compress neem cover images

kidSoba committed:
refactor: add optional compression value to image-compression function

kidSoba committed:
refactor: change console output for downloading cover images

kidSoba committed:
feat: add function to download and compress all markdown images

kidSoba committed:
refactor: change path for cover-images

kidSoba committed:
feat: toggle off fetching for neem-overview stuff in DEBUG-mode

kidSoba committed:
refactor: update overview.zip-contents

kidSoba committed:
fix: remove bug where container would crash after stopping and restarting with default neem-data

When the container was started in debug-mode, then stopped and restarted (all while in debug-mode), then the restart would make the container crash. The container tries to copy certain contents of the overview.zip, but these will already exist, so the directory needs to be removed before copying.

kidSoba committed:
feat: add page-links for neem-homepage etc. to overview pages

kidSoba committed:
feat: add css-styling for overview-pages

Commit:
Fix host error by removing ports

Commit:
Merge branch 'footer-and-legal-pages' of github.com:navidJadid/openease

Commit:
first version of openease rules that works with mongolog

kidSoba committed:
fix: remove publications-links for neem-cards on homepage and on overview pages

kidSoba committed:
style: add linebreaks for shorter lines

kidSoba committed:
refactor: remove links to specific publications

kidSoba committed:
refactor: refactor utility.py and sections which use refactored code

kidSoba committed:
refactor: improve code for overview-page redirect

kidSoba committed:
refactor: remove unneeded imports

kidSoba committed:
refactor: use utility.py unzip function

Commit:
Merge pull request #99 from sasjonge/mongolog_fix

Added fixes so that the rules in openEASE works with mongolog

Commit:
Merge pull request #98 from sasjonge/master

Fix host error with newer docker-compose versions

kidSoba committed:
refactor: refactor code for overview pages

kidSoba committed:
refactor: give external links on homepage noreferrer and noopener

kidSoba committed:
feat: add (temporary) news section

kidSoba committed:
feat: add publications pages

default_papers needs to be split due to GitHub file size limit

kidSoba committed:
fix: set publications update interval to every day instead of every 3h

sasjonge commented:
The base path so for example https://localhost should go to the homepage and not to the login page. Also it would be good to add potential cronjobs etc. into the README

kidSoba committed:
refactor: remove unnecessary whitespace

kidSoba committed:
refactor: set homepage-route to "/"

kidSoba committed:
refactor: add check for publications data for single-publications page

kidSoba committed:
fix: add check for publications-page to check if papers-directory is empty

kidSoba committed:
fix: add check for requested paper for download

kidSoba committed:
feat: remove paper-zips, add check when loading defaults and add option for download

kidSoba committed:
fix: minor whitespace issue

kidSoba committed:
refactor: refactor some constants and paths, also move some function to utility.py

kidSoba committed:
refactor: use methods from utility.py

kidSoba committed:
fix: add check for bibtex-file to not replace after download if pybtex throws an error

kidSoba committed:
fix: fix faulty check and refactor a little

kidSoba committed:
refactor: reorder some lines

kidSoba committed:
refactor: move and rename some constants

kidSoba committed:
refactor: move path constants to settings.py

kidSoba committed:
refactor: change several paths

kidSoba committed:
fix: adjust base.html url to homepage

kidSoba committed:
fix: adjust wrong path for markdowns

kidSoba committed:
feat: add copy functions to utility.py

kidSoba committed:
refactor: remove unnecessary import

kidSoba committed:
refactor: refactor some imports to use functions from utility.py

kidSoba committed:
refactor: remove unnecessary imports from overview.py

kidSoba committed:
feat: add downloadable files for publications and overview for admin-settings

kidSoba committed:
refactor: remove unneeded imports

kidSoba committed:
refactor: add ids to background-scheduler jobs

kidSoba committed:
feat: add mutex-lock wrapper and make some update functions mutex-locked

kidSoba committed:
refactor: remove unnecessary whitespace

kidSoba committed:
refactor: add more robustness for download method

instead of crashing the whole program, failed downloads of neem markdowns are now caught in a try-except block

kidSoba committed:
refactor: extract method for cleaner code

kidSoba committed:
fix: replace normal threading with ThreadPoolExecutor

due to certain details of the Tornado framework, the threading module would block the whole module during execution; the ThreadPoolExecutor bypasses this problem

kidSoba committed:
fix: add more robustness to the content-update functions

proper back-up methods are supplied that will not block the thread

kidSoba committed:
refactor: check if dir exists before creating it

kidSoba committed:
refactor: change order some methods are executed and adjust some comments

kidSoba committed:
refactor: restructure some code & remove unneeded functions and comments

kidSoba committed:
feat: make scheduler tasks coalesce in case of missed execution

kidSoba committed:
feat: set fixed start time for publications-update job

kidSoba committed:
feat: add function to manually update or load content in admin settings

kidSoba committed:
refactor: rename overview.py to neem_overview.py

kidSoba committed:
doc: add comment for download functions

kidSoba committed:
refactor: force download-file function to write to temp folder first, then move the file

kidSoba committed:
refactor: change some line comments to block comments

kidSoba committed:
refactor: move backgroundscheduler into own module + some small stuff

background scheduler is now in its own module; also scheduler jobs now coalesce,
have IDs, and the publications job is scheduled to 3 a.m. EST

kidSoba committed:
fix: add fail-safe for papers unzip

this is necessary if download_default_papers is set to True but the download
of the file failed and therefore is not in the expected location

kidSoba committed:
fix: correct the wrong check for background-scheduler state

kidSoba committed:
refactor: move several settings to postgres-sql

this lets settings persist over restarts and rebuilds

kidSoba committed:
refactor: move download-dir creation into own method

kidSoba committed:
refactor: change method parameter name

kidSoba committed:
refactor: update set-methods to have less redundant code

kidSoba committed:
refactor: make get_first_entry private

kidSoba committed:
refactor: move utility.py to helpers-dir

kidSoba committed:
fix: change condition for preparing download files for admin

kidSoba committed:
refactor: move several routing functions into the proper modules

kidSoba committed:
refactor: adjust whitespace for better style

kidSoba committed:
fix: fix error with retrieving neemhub-settings on fresh installs

due to unfortunately placed imports, the container would crash on fresh
installs trying to fetch neemhub-settings from postgres, even though
tables hadn't been created yet

kidSoba committed:
feat: extend content settings and update init_app()

kidSoba committed:
feat: add feature to fetch next job runtime from background-scheduler

kidSoba committed:
feat: set content-settings accordingly when necessary

methods that update or load certain content files will now also update the
content-settings in the database to match the current state

kidSoba committed:
refactor: rename content update functions

kidSoba committed:
refactor: move constant to module-level

kidSoba committed:
fix: loading default-files will reset last-update time

kidSoba committed:
feat: update layout of content-settings-page

kidSoba committed:
feat: add missing fields and adjust page-layout

kidSoba committed:
refactor: move code-snippet

kidSoba committed:
refactor: remove unneeded whitespace

kidSoba committed:
feat: add utility method to get obj-type as str

kidSoba committed:
refactor: rename ContentState enum-class member

kidSoba committed:
fix: address bug with writing to content-settings db

The init-methods for the db were assigning to the class itself, not the
instance. Therefore there was an issue where the values assigned to in
the init_method could not be written to in the db.

kidSoba committed:
feat: add check for datatype in db

kidSoba committed:
feat: add url-fields and respective set-methods to db

kidSoba committed:
feat: add color-scheme for flash of successful and unsuccessful tasks

kidSoba committed:
refactor: rename and adjust publications-download-zip-path constant

kidSoba committed:
refactor: adjust logger messages for preparing downloadable files

kidSoba committed:
fix: add missing function brackets

kidSoba committed:
fix: add missing db update for content update-method

kidSoba committed:
fix: add check for whether a neem has a md before adding to recent-neems

kidSoba committed:
refactor: adjust info-message

kidSoba committed:
fix: redirect to correct url instead of calling render method

this would cause the url to stay the same, which could cause problems on reloads

kidSoba committed:
refactor: refactor update-content methods to offer
"automatic" and "manual" version for other modules
to use

kidSoba committed:
fix: add logger to background-scheduler

necessary because if not done so, it might crash the program

kidSoba committed:
refactor: change scheduler trigger-type to "cron"

kidSoba committed:
faet: add remaining content-settings

kidSoba committed:
refactor: use db-config values for various tasks

kidSoba commented:
Previous Commits include:

- various bug-fixes and refactorings
- admin content-settings page is now usable (/settings/content) and offers the following settings, which can be viewed and adjusted:
    - turn on/off downloading default papers
    - turn on/off preparing downloadable files for admin
    - load defaults for content categories
    - manually load updates for content categories (instead of waiting for the server to fetch updates)
    - pause/resume content-update jobs
    - download currently loaded content-files
    - enter urls for publications.bib, default_papers.zip, papers.zip
- localhost/ now goes to the homepage

README will be updated soon. Tooltips shall be added as well.

kidSoba committed:
feat: add function to check if string is valid url

kidSoba committed:
refactor: check if strings are urls before assigning to url-fields

kidSoba committed:
fix: check if publications url is set before trying to fetch updates

kidSoba committed:
feat: deactivate certain buttons in settings if another setting is not yet set

kidSoba committed:
refactor: change flash categories

kidSoba committed:
fix: change display property from block to table-row

kidSoba committed:
feat: add tooltips to their correct position in content-settings

actual text has not been added yet

kidSoba committed:
feat: change location of tooltip to end of line

kidSoba committed:
fix: set download buttons to disabled if setting is set accordingly

kidSoba committed:
fix: change when to disable downloadable files

files can now only be downloaded if the file is available

kidSoba committed:
refactor: add tooltips texts for whole content-settings page

kidSoba committed:
docs: add information about content stuff

kidSoba committed:
docs: mention detail for urls on content-settings page

Commit:
fixed bug if directly starting the tutorial page

Commit:
Merge pull request #102 from sasjonge/master

Fixed bug if directly starting the tutorial page

Commit:
feat: add download of openease js-modules to dockerfile

kidSoba committed:
fix: fix brackets around ARG-references

kidSoba committed:
fix: remove sections that would brick the docker build

kidSoba committed:
Merge branch 'ease-crc:master' into dockerfile-node-module-refactor

kidSoba committed:
fix: add back the feature to use local node-modules for development

kidSoba committed:
Merge branch 'dockerfile-node-module-refactor' of https://github.com/navidJadid/openease into dockerfile-node-module-refactor

Commit:
Merge pull request #103 from navidJadid/dockerfile-node-module-refactor

Dockerfile node module refactor

kidSoba committed:
fix: replace package.json git-urls with https-url

Due to security changes this is a necessity

kidSoba committed:
fix: update npm-dependies which had vulnerabilities

Commit:
Merge pull request #104 from navidJadid/package-json-fix

Update npm-dependencies with known vulnerabilities

kidSoba committed:
refactor: remove unnecessary whitespace

kidSoba committed:
docs: update node_modules README

kidSoba committed:
refactor: improve side-card appearance

kidSoba committed:
feat: featured neems now appear in a carousel

kidSoba committed:
style: adjust some spacings

kidSoba committed:
fix: improve neemhub page appearance

moved pagination links to center and reduced number of neems per page to 8

kidSoba committed:
fix: remove publication paragraph from homepage

kidSoba committed:
fix: remove tutorials paragraph from homepage

kidSoba committed:
fix: slightly adjust opening paragraph and move video to top

kidSoba committed:
fix: adjust homepage column layout

kidSoba committed:
fix: improve neem-card appearance, especially for touch devices

kidSoba committed:
fix: move software-components paragraph on homepage

kidSoba committed:
refactor: move file operations to own module

kidSoba committed:
refactor: move threading and mutex to own module

kidSoba committed:
fix: change module name to avoid name collision

kidSoba committed:
refactor: remove unnecessary code

kidSoba committed:
refactor: rename variable for better readability

kidSoba committed:
refactor: move global variable to local level

kidSoba committed:
fix: add clause to template to check whether dicts have elements

kidSoba committed:
refactor: simplify expression

kidSoba committed:
style: remove useless whitespace

kidSoba committed:
style: format default files

kidSoba committed:
refactor: rename constant

kidSoba committed:
refactor: change paths for constants

kidSoba committed:
refactor: rename get_sanitizer

kidSoba committed:
refactor: remove unnecessary function

kidSoba committed:
docs: move comment block

kidSoba committed:
fix: correct condition for if statement

kidSoba committed:
refactor: change file existence check to own method

kidSoba committed:
fix: papers download now checks if url is valid

kidSoba committed:
refactor: change directory existence check to own method

kidSoba committed:
refactor: move file deletion to own method

kidSoba committed:
refactor: move deletion of non-empty dirs to own method

kidSoba committed:
refactor: move deletion of empty dirs to own method

kidSoba committed:
refactor: use own method for checking if path is a dir

kidSoba committed:
refactor: move get file extension to own method

kidSoba committed:
refactor: move get path name to own method

kidSoba committed:
refactor: move get path stem to own method

kidSoba committed:
refactor: move get path parent to own method

kidSoba committed:
refactor: move check if dir has any items to own method

kidSoba committed:
refactor: move make dir to own method

kidSoba committed:
fix: download needs to overwrite files

kidSoba committed:
feat: add db-table helper functions

kidSoba committed:
fix: change enum orders

kidSoba committed:
feat: add helper functions for counting dir items

kidSoba committed:
fix: change path of downloads_dir

kidSoba committed:
refactor: use other function

kidSoba committed:
feat: migrate publications and neem-overview data to db

Publications and neem-overview now use the database for persistent storage instead of a proprietary solution.  Other necessary files are stored in volumes, which reduces start -up  time, since files do not need to be reloaded on each build or start.
